ui/src/views/user/signup/SignupView.vue

```diff
@@ -112,8 +114,43 @@ export default {
   },
   methods: {
     onFormSubmit() {
-      // to do: post form data to back-end's registration api
-      this.$router.push({ name: "general-user-signup-thankyou" });
+      userSignup(this.formData)
+        .then((response) => {
+          if (response["message"].includes(" already exists.")) {
+            this.$notify.toast(response["message"]);
+          }
+          else {
+            this.$router
+              .push({ name: "general-user-signup-thankyou" })
+              .catch(() => {});
+          }
+        })
+        .catch((error) => {
```

Comment:
Rather than putting the error message here, we can make use of the interceptor XD    I think I have put a basic interceptor there. It will be great if you can move the notification there, as well as  importing vuetify-notify there... but I dont remember if vuetify-notify can be exported or no XD   the package was as minimal as it can be. 

But if it's too much work, then this is OK.

Reply:
Yes, there is a basic interceptor, but I'm not sure how to import vuetify-notify into ui/src/views/user/signup/SignupView.vue correctly. I suppose simply putting `this.$notify.toast("Something went wrong. Please try again later.");` inside the interceptor without importing vuetify-notify won't do it. And I'm afraid testing different ways to do that will spend way too much time (especially because re-building the ui service after every code change is really slow).

Therefore, I am leaving the error message as it is now, so that I can start working on the other API calls (I intend to work on user login next).

Reply:
It wont do because vuetify-notify is part of the Vue environment, importing it will be a bit tricky XD so lets leave it for now.